graph/push_v2.go

```diff
@@ -141,6 +145,10 @@ func (p *v2Pusher) pushV2Tag(tag string) error {
 		dgst, err := p.graph.GetDigest(layer.ID)
 		switch err {
 		case nil:
+			if p.layersPushed[layer.ID] {
+				exists = true
+				break
```

Comment:
The check really is nested in the `case nil:` since `exists` can only be `true`  if received a valid digest, otherwise we will need to push to at least get the digest of the layer for the manifest. Using a label here would be gross when it could be achieved with a comment.